src/cocoa/toga_cocoa/window.py

```diff
@@ -242,3 +242,9 @@ def stack_trace_dialog(self, title, message, content, retry=False):
 
     def save_file_dialog(self, title, suggested_filename, file_types):
         return dialogs.save_file(self.interface, title, suggested_filename, file_types)
+
+    def open_file_dialog(self, title, initial_directory, file_types, multiselect):
+        return dialogs.open_file(self.interface, title, file_types, multiselect)
+
+    def select_folder_dialog(self, title, initial_directory):
```

Comment:
@danyeaw @freakboy3742 Offline discussion w/ @nehararora interface for selecting folder does not  support multiselect, but in macOS, because selecting files and selecting folders uses the same component, you can select multiple folders in macOS. Should we change the interface and add in an optional arg multiselect=False, or should we not expose that if macOS is a one-off (I'm not sure if it is)?

Reply:
Offline discussion concluded interface should support all feature flags required by all backends; otherwise functionality would be hidden from one backend to another. Updating interface for selecting a folder.

Reply:
Offline discussion w/ @nehararora @danyeaw @freakboy3742 beeware should support the intersection of functionality and not the union of functionality, unless there is a way in order to model the problem to support the same functionality across all different backends so that user code does not need to change. In the case of multiselect, update the method signature from interfaces to backends to support optional argument multiselect=False, and if the backend does not support multiselect for folders, return a list of folders that will always be of length 0 (if no folders selected) or 1 (max one folder can be selected), so that the type returned will be the same across all backends.